src/utils/auth_utils.js

```diff
@@ -22,22 +28,70 @@ const firebaseConfig = {
 const app = initializeApp(firebaseConfig);
 const auth = getAuth(app);
 
-const signInWithGoogle = async () => {
-  // TODO
-};
-
 const logInWithEmailAndPassword = async (email, password) => {
   try {
     await signInWithEmailAndPassword(auth, email, password);
     alert('Sign in success');
   } catch (err) {
-    console.error(err);
     alert(err.message);
   }
 };
 
-const registerWithEmailAndPassword = async () => {
-  // TODO
+const createUserInFirebase = async (email, password) => {
+  try {
+    const user = await createUserWithEmailAndPassword(auth, email, password);
+    sendEmailVerification(user.user);
+    return user;
+  } catch (err) {
+    throw new Error(err.message);
+  }
+};
+
+const createUserInDB = async userObject => {
+  try {
+    // Replace line below with call to NPO DB
+    // const res = await WMKBackend.post('/register/create', userObject);
+    // return res;
+  } catch (err) {
+    const { email, password } = userObject;
+
+    // Since this route is called after user is created in firebase, if this
+    // route errors out, that means we have to discard the created firebase object
+    await signInWithEmailAndPassword(auth, email, password);
+    const userToBeTerminated = await auth.currentUser;
+    userToBeTerminated.delete();
+
+    throw new Error(err.message);
+  }
+};
+
+const createUser = async (email, password) => {
+  try {
+    await createUserInFirebase(email, password);
+    createUserInDB({ email, password });
+  } catch (err) {
+    throw new Error(err.message);
```

Comment:
Don't need try/catch because will be caught from a parent function and we are not doing anything when catching the error here other than throwing it again.